app/code/core/Mage/Core/Model/Resource/Session.php

```diff
@@ -185,10 +183,9 @@ public function close()
      * Fetch session data
      *
      * @param string $sessId
-     * @return string
+     * @return string|false
      */
-    #[\ReturnTypeWillChange]
-    public function read($sessId)
+    public function read($sessId): string|false
```

Comment:
line 201:
```php
         return (string)$data;
```
to reflect the return type, it should be

```php
        return $data;
```

After the change, you can test with
```php
$value = Mage::getResourceSingleton('core/session')->read('abc'); // $value is false
```

Reply:
done, although the db->fetchOne() is marked to return a string...